sleepy/types.py

```diff
@@ -436,20 +439,20 @@ class UnionType(Type):
   A tagged union, i.e. a type that can be one of a set of different types.
   """
 
-  def __init__(self, possible_types: List[Type], possible_type_nums: List[int], val_size: Optional[int],
+  def __init__(self, type_mapping: Mapping[Type, int],
```

Comment:
I think `type_mapping` is not a very descriptive name. Maybe `variants_map`? `possible_types_map`?